js-exercises/counting-anagrams/countingAnagrams.js

```diff
@@ -0,0 +1,26 @@
+const countingAnagrams = (str) => {
+    const tempAnagrams = {}
+    let flag = 0
+    str.forEach((loopingArrayWord)=>{
+      const wordAfterSorting = returnTheWordInSorted(loopingArrayWord)
+      if(tempAnagrams[wordAfterSorting])
+      {
```

Comment:
From line 4 to 13  you are doing lots of intermixing of logics .. that can be separated. and that making it hard to read/ understand.

And its not passing all test cases as well
![image](https://user-images.githubusercontent.com/17058535/77518064-1ccad180-6ea3-11ea-9e5d-ec5d66408f62.png)